tests/integration/test_sqs.py

```diff
@@ -361,6 +361,33 @@ def test_batch_send_with_invalid_char_should_succeed(self, sqs_client, sqs_creat
         result_send = sqs_client.send_message_batch(QueueUrl=queue_url, Entries=batch)
         assert len(result_send["Failed"]) == 1
 
+    @pytest.mark.skipif(
+        os.environ.get("TEST_TARGET") == "AWS_CLOUD", reason="Testing LocalStack specific config"
+    )
+    def test_external_hostname(self, monkeypatch, sqs_client, sqs_create_queue):
+        external_host = "external-host"
+        external_port = "12345"
+        SQS_PORT_EXTERNAL = "SQS_PORT_EXTERNAL"
+
+        monkeypatch.setattr(config, SQS_PORT_EXTERNAL, external_port)
+        monkeypatch.setattr(config, "HOSTNAME_EXTERNAL", external_host)
+        # TODO: remove once the old provider is discontinued
+        from localstack.services.sqs import sqs_listener as old_sqs_listener
+
+        monkeypatch.setattr(old_sqs_listener, SQS_PORT_EXTERNAL, external_port)
+
+        os.environ[SQS_PORT_EXTERNAL] = external_port
```

Comment:
nitpick: Let's better use `monkeypatch.setenv(..)` here (to make sure the env gets reset after the test is finished)

Reply:
whoops missed that one, thanks!